tests/make/cfn_lint_failure.bats

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bats
+
+TEST_DIR="$(pwd)/cfn_lint_failure"
+
+# generate cfn templates in nested folders
+function setup() {
+rm -rf "$TEST_DIR"
+working_dirs=("$TEST_DIR" "$TEST_DIR/nested")
+for working_dir in "${working_dirs[@]}"
+do
+
+  mkdir -p "$working_dir"
+  # https://s3.amazonaws.com/cloudformation-templates-us-east-1/S3_Website_Bucket_With_Retain_On_Delete.template
+  cat > "$working_dir/test.template.cfn.json" <<"EOF"
+{
+  "AWSTemplateFormatVersion" : "2010-09-09",
+
+  "Description" : "AWS CloudFormation Sample Template S3_Website_Bucket_With_Retain_On_Delete: Sample template showing how to create a publicly accessible S3 bucket configured for website access with a deletion policy of retain on delete. **WARNING** This template creates an S3 bucket that will NOT be deleted when the stack is deleted. You will be billed for the AWS resources used if you create a stack from this template.",
+
+  "Resources" : {
+    "S3Bucket" : {
+      "Type" : "AWS::S3::Bucket",
+      "Properties" : {
+        "AccessControl" : "PublicRead",
+        "WebsiteConfiguration" : {
+          "IndexDocument" : "index.html",
+          "ErrorDocument" : "error.html"
+        }
+      },
+      "DeletionPolicy" : "Retain",
+    }
+  },
+
+  "Outputs" : {
+    "WebsiteURL" : {
+      "Value" : { "Fn::GetAtt" : [ "S3Bucket", "WebsiteURL" ] },
+      "Description" : "URL for website hosted on S3"
+    },
+    "S3BucketSecureURL" : {
+      "Value" : { "Fn::Join" : [ "", [ "https://", { "Fn::GetAtt" : [ "S3Bucket", "DomainName" ] } ] ] },
+      "Description" : "Name of S3 bucket to hold website content"
+    }
+  }
+}
+EOF
+
+  cat > "$working_dir/test.template.cfn.yaml" <<"EOF"
+AWSTemplateFormatVersion: 2010-09-09
+Description: >-
+  AWS CloudFormation Sample Template S3_Website_Bucket_With_Retain_On_Delete:
+  Sample template showing how to create a publicly accessible S3 bucket
+  configured for website access with a deletion policy of retain on delete.
+  **WARNING** This template creates an S3 bucket that will NOT be deleted when
+  the stack is deleted. You will be billed for the AWS resources used if you
+  create a stack from this template.
+Resources:
+  S3Bucket:
+    Type: 'AWS::S3::Bucket'
+    Properties:
+      AccessControl: PublicRead
+      WebsiteConfiguration:
+        IndexDocument: index.html
+        ErrorDocument: error.html
+    DeletionPolicy: Retain
+Outputs:
+  WebsiteURL:
+    Value: !GetAtt
+      - S3Bucket
+      - WebsiteURL
+    Description: URL for website hosted on S3
+  S3BucketSecureURL:
+    Value: !Join
+      - ''
+      - - 'https://'
```

Comment:
Is this supposed to be a double dash --?

Reply:
probably, but I'm opting to leave this as is given that its a straight copy paste of the source template after being converted from json to yaml in the aws cfn template designer.

Reply:
See I was actually thinking that is was supposed to be a single dash and the werid double dash was just a copy pasta error but if I am fine with it staying that was if it's from the template designer.